chat/test/handlers/test_opensearch_neural_search.py

```diff
@@ -0,0 +1,43 @@
+# ruff: noqa: E402
+import sys
+sys.path.append('./src')
+
+from unittest import TestCase
+from handlers.opensearch_neural_search import OpenSearchNeuralSearch
+from langchain_core.documents import Document
+
+class MockClient():
```

Comment:
Is there a way to make this a real mock/spy and make some assertions about the body received by the search method? Or would that be going overboard?

Reply:
I was going for the minimal solution angle here, but I'm open to other ideas if you've got them! Maybe in a separate issue though, just to keep this one moving?